### DISCUSSION TITLE
polling and publish interval

### DISCUSSION THREAD

**Discussion by rvicev** [tbnobody/OpenDTU#1180] [Q&A]
The default inverter polling and mqtt publish intervals are 5 seconds.

What would be the minimum inverter and mqtt publish intervals, using an Olimex PoE ESP and CMT2300A radio board ?

(I understand 5 seconds should be more than sufficient for reporting data; however in certain instances, such as power limiting, a shorter interval may seem to be of interest)

**Comment by rvicev** [comment]
Is there really nobody out there who has tried setting these intervals to 4 seconds or less ???

**Reply by Solarteur** [reply]
You don't "buy" a ready-made solution with OpenDTU, you have to take care of specific questions yourself: What prevents you from testing different polling intervals for your specific system (e.g. number of inverters)?

**Comment by rvicev** [comment]
Thank you, I know all that. Obviously I haven´t got my system yet, otherwise I would have tried it already without asking ...

I am still in the planning phase of an offgrid system (Victron MultiPlus II 5 kVa + 6x HMT-800 inverters  on AC-OUT1, + battery). In an example given by Victron they have shown that, when the battery is almost full, by using frequency shifting with a Fronius they can bring down the output of the Fronius to (close to) zero in just under 2 minutes.

Frequency shifting may be reliable (and is also supported by Hoymiles), but it also has certain disadvantages.

2 minutes seems a reasonable amount of time, so I was wondering if this would be possible with Hoymiles without frequency shifting, using OpenDTU / MQTT. This would of course also require a sufficiently large battery, which would give sufficient time preventing the battery voltage from rising too quickly.

Therefor I am interested to know if anybody has tested if the ESP32 can handle 4, 3 or 2 seconds polling / publish intervals.
(I have zero experience with ESP32, so no idea how powerful it is)
